sdk/core/System.ClientModel/src/Convenience/ApiKeyCredential.cs

```diff
@@ -24,14 +29,33 @@ public class ApiKeyCredential
     public ApiKeyCredential(string key) => Update(key);
 #pragma warning restore CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable.
 
-    // Note: this is uses a method instead of a Key property so if an instance of
-    // the type were passed to a serializer that serializes public properties
-    // it doesn't inadvertently expose a secret.  We call it Deconstruct to allow
-    // consistent naming across credential types.  For example, for
-    // NamedKeyCredential, we expect to have a Deconstruct method with two out
-    // params to enable atomicity in reading multiple credential values.
+    // This type uses a method instead of a Key property so that, if an instance
+    // of the type were passed to a serializer that serializes public properties
+    // by default, it doesn't inadvertently expose a secret. The method has the
+    // name Deconstruct to allow consistent naming across credential types.  For
+    // example, a corresponding NamedKeyCredential type will have a Deconstruct
+    // method with two out params to make reading multiple credential values an
+    // atomic operation.
+
+    /// <summary>
+    /// Deconstructs the credential into component values.
+    /// </summary>
+    /// <param name="key">When this method returns, contains the value of the
+    /// API key this instance represents.</param>
     public void Deconstruct(out string key) => key = Volatile.Read(ref _key);
 
+    /// <summary>
+    /// Updates the service key.
```

Comment:
```suggestion
    /// Updates the API key used to authenticate with the service.
```

nit: We should probably be consistent in how we refer to the key.  We've used "API key" in most other places.   Calling it "service key" here may be confusing.